packages/utils/src/validators.ts

```diff
@@ -6,7 +6,46 @@ import {
   getNamespacesEventsForChainId,
   getAccountsChains,
 } from "./namespaces";
-import { hasOverlap, getDeletedDiff } from "./misc";
+import { getSdkError, getInternalError } from "./errors";
+import { hasOverlap } from "./misc";
+
+export type ErrorObject = { message: string; code: number } | null;
+
+// -- types validation ----------------------------------------------------- //
+
+export function isValidArray(arr: any, itemCondition?: (item: any) => boolean) {
+  if (Array.isArray(arr)) {
+    if (typeof itemCondition !== "undefined" && arr.length) {
+      const matches = arr.filter(itemCondition);
+      return matches.length === arr.length;
+    } else {
+      return true;
+    }
+  }
+  return false;
+}
+
+export function isValidObject(obj: any) {
+  return Object.getPrototypeOf(obj) === Object.prototype && Object.keys(obj).length;
+}
+
+export function isUndefined(input: any): input is undefined {
+  return typeof input === "undefined";
+}
+
+export function isValidString(input: any, optional: boolean) {
+  if (optional && isUndefined(input)) return true;
+
+  return typeof input === "string" && Boolean(input.trim().length);
+}
+
+export function isValidNumber(input: any, optional: boolean) {
+  if (optional && isUndefined(input)) return true;
+
+  return typeof input === "number";
```

Comment:
Another "fun" little JS nugget, we need to guard against `NaN` here because:

```
typeof NaN
-> 'number'
```

```suggestion
  return typeof input === "number" && !isNaN(input);
```

---

(😂 😭 )

Reply:
Lol I just realised this code just moved up in the file 🤦 

Anyways good to double check I guess 🤷